### PR TITLE
[receiver/postgresql] Add bgwriter Metrics

### LINKED ISSUE #13327: [receiver/postgresql] Add `bgwriter` Metrics
**Is your feature request related to a problem? Please describe.**
Adding more metrics initially spun up from #12338 

These are the `bgwriter` metrics related to the background process introduced in postgres 8.0 that occasionally flushes modified pages out of the shared buffer pool.

More information on the source of these metrics can be found [here](https://www.postgresql.org/docs/current/monitoring-stats.html).

Planning on putting this behind the feature gate: `receiver.postgresql.emitMetricsWithResourceAttributes`

**Describe the solution you'd like**
Metrics proposing to add:
```
postgresql.bgwriter.buffers.allocated
postgresql.bgwriter.buffers.writes
postgresql.bgwriter.checkpoint.count
postgresql.bgwriter.duration
postgresql.bgwriter.maxwritten.count
```

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by schmikei**
**Description:** Adds background writer metrics for the `postgresqlreceiver`. Feature is behind the `receiver.postgresql.emitMetricsWithResourceAttributes` feature gate. 

Metrics Added
- postgresql.bgwriter.buffers.allocated
- postgresql.bgwriter.buffers.writes
- postgresql.bgwriter.checkpoint.count
- postgresql.bgwriter.duration
- postgresql.bgwriter.maxwritten.count

**Link to tracking Issue:** Resolves #13327

**Testing:** Scraper Testing and Integration Testing

**Documentation:** Metric documentation in `documentation.md`

### CODE REVIEW

File: receiver/postgresqlreceiver/metadata.yaml [hunk 1]
```diff
@@ -37,6 +57,52 @@ attributes:
     enum: [dead, live]
 
 metrics:
+  postgresql.bgwriter.buffers.allocated:
+    attributes: []
+    description: Number of buffers allocated.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/metadata.yaml:66]: Just double checking - this isn't the real time number of buffers allocated, right?

File: receiver/postgresqlreceiver/metadata.yaml [hunk 2]
```diff
@@ -37,6 +57,52 @@ attributes:
...
+      value_type: int
+    unit: "{buffers}"
+  postgresql.bgwriter.buffers.writes:
+    attributes:
+      - bg_buffer_source
+    description: Number of buffers written
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/metadata.yaml:72]:
```suggestion
    description: Number of buffers written.
```

File: receiver/postgresqlreceiver/metadata.yaml [hunk 3]
```diff
@@ -37,6 +57,52 @@ attributes:
...
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: "{buffers}"
+  postgresql.bgwriter.checkpoint.count:
+    attributes:
+      - bg_checkpoint_type
+    description: The number of checkpoints performed
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/metadata.yaml:82]:
```suggestion
    description: The number of checkpoints performed.
```

File: receiver/postgresqlreceiver/metadata.yaml [hunk 4]
```diff
@@ -37,6 +57,52 @@ attributes:
...
+    attributes:
+      - bg_duration_type
+    description: Total time spent writing and syncing files to disk by checkpoints.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: ms
+  postgresql.bgwriter.maxwritten.count:
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/metadata.yaml:99]:
```suggestion
  postgresql.bgwriter.maxwritten.count:
```

I think this name needs tweaking. The namespace here is `postgresql.bgwriter.maxwritten`. Typically the last word of a namespace is a noun, and the metric describes the noun in some way. Maybe just `postgresql.bgwriter.maxwritten` is appropriate here?

**Comment by schmikei** [comment]
@djaglowski do you mind taking a look at this one?

**Review by djaglowski — COMMENTED** [review]
Looks good, just a couple questions

File: receiver/postgresqlreceiver/metadata.yaml (hunk 1 above)
Comment (schmikei) [reply receiver/postgresqlreceiver/metadata.yaml:66]: The documentation is a tad spotty, I believe it is a normal counter based off local testing. I also found this discussion that seems to indicate it being a normal counter https://www.postgresql.org/message-id/1a00e942-e7c6-8e4d-afec-4b1befd5688d%40pro-open.de

**Review by djaglowski — APPROVED** [review]

**Pull request merged**